src/ow1/heroes/mccree.opy

```diff
@@ -37,4 +37,22 @@ rule "[mccree.opy]: Stun flashed enemy":
     @Condition eventAbility == 0
     @Condition attacker.getCurrentHero() == Hero.MCCREE or attacker.getCurrentHero() == Hero.GENJI
 
-    victim.setStatusEffect(eventPlayer, Status.STUNNED, OW1_MCCREE_FLASHBANG_STUN_DURATION)
\ No newline at end of file
+    victim.setStatusEffect(eventPlayer, Status.STUNNED, OW1_MCCREE_FLASHBANG_STUN_DURATION)
+
+rule "[mccree.opy]: Remove damage reduction during combat roll":
+    @Event eachPlayer
+    @Hero mccree
+    @Condition eventPlayer.isUsingAbility1() == true
+    
+    startDamageModification(eventPlayer, getAllPlayers(), 200, DamageReeval.RECEIVERS_DAMAGERS_AND_DMGPERCENT)
+    waitUntil(not eventPlayer.isUsingAbility1(), 0.5)
+    stopDamageModification(getLastDamageModification())
+
+rule "[mccree.opy]: Remove damage reduction during ultimate":
+    @Event eachPlayer
+    @Hero mccree
+    @Condition eventPlayer.isUsingUltimate() == true
+    
+    startDamageModification(eventPlayer, getAllPlayers(), 180, DamageReeval.RECEIVERS_DAMAGERS_AND_DMGPERCENT)
+    waitUntil(not eventPlayer.isUsingUltimate(), 7.1)
+    stopDamageModification(getLastDamageModification())
```

Comment:
There's a non-negligible chance that the last damage modification changes to a different damage modification event during the wait, which would make Cassidy permenantly take 180% more damage.